### ISSUE TITLE
6.keras_CNN_Cifar10.ipynb

### ISSUE THREAD

**Issue by Hank666818** [Hank666818/-#4]
from tensorflow import keras
from keras.models import Sequential
from keras.layers import Dense, Conv2D, MaxPooling2D, Flatten
model=Sequential()
model.add(Conv2D(filters=64, kernel_size=(5, 5), input_shape=(32, 32, 3), activation='relu'))
model.add(Conv2D(filters=64, kernel_size=(5, 5), activation='relu'))
model.add(MaxPooling2D(pool_size=2))
model.add(Conv2D(filters=128, kernel_size=(5, 5), activation='relu'))
model.add(Conv2D(filters=128, kernel_size=(5, 5), activation='relu'))
model.add(MaxPooling2D(pool_size=2))
model.add(Flatten())
model.add(Dense(10, activation='softmax'))
model.summary()
model.compile(optimizer='rmsprop', loss='categorical_crossentropy', metrics=['accuracy'])
     
Model: "sequential"
_________________________________________________________________
 Layer (type)                Output Shape              Param #   
=================================================================
 conv2d (Conv2D)             (None, 28, 28, 64)        4864      
                                                                 
 conv2d_1 (Conv2D)           (None, 24, 24, 64)        102464    
                                                                 
 max_pooling2d (MaxPooling2D  (None, 12, 12, 64)       0         
 )                                                               
                                                                 
 conv2d_2 (Conv2D)           (None, 8, 8, 128)         204928    
                                                                 
 conv2d_3 (Conv2D)           (None, 4, 4, 128)         409728    
                                                                 
 max_pooling2d_1 (MaxPooling  (None, 2, 2, 128)        0         
 2D)                                                             
                                                                 
 flatten (Flatten)           (None, 512)               0         
                                                                 
 dense (Dense)               (None, 10)                5130      
                                                                 
=================================================================
Total params: 727,114
Trainable params: 727,114
Non-trainable params: 0
_________________________________________________________________

cifar10=keras.datasets.cifar10
(train_data, train_label), (test_data, test_label) = cifar10.load_data()
train_label_original=train_label
train_label = keras.utils.to_categorical(train_label)
test_label = keras.utils.to_categorical(test_label)
     

import matplotlib.pyplot as plt
category=['airplane','automobile','bird','cat','deer','dog','frog','horse','ship','truck']
plt.imshow(train_data[0])
plt.title(category[int(train_label_original[0])])
plt.show()
print(train_label_original[0])
     

[6]

train_history=model.fit(train_data,train_label,epochs=30,batch_size=512)

     
Epoch 1/30
98/98 [==============================] - 12s 54ms/step - loss: 21.2404 - accuracy: 0.1729
Epoch 2/30
98/98 [==============================] - 4s 44ms/step - loss: 2.3775 - accuracy: 0.2428
Epoch 3/30
98/98 [==============================] - 4s 45ms/step - loss: 2.1899 - accuracy: 0.2945
Epoch 4/30
98/98 [==============================] - 4s 45ms/step - loss: 1.8240 - accuracy: 0.3722
Epoch 5/30
98/98 [==============================] - 4s 45ms/step - loss: 1.7623 - accuracy: 0.3973
Epoch 6/30
98/98 [==============================] - 4s 45ms/step - loss: 1.5814 - accuracy: 0.4564
Epoch 7/30
98/98 [==============================] - 4s 45ms/step - loss: 1.5443 - accuracy: 0.4749
Epoch 8/30
98/98 [==============================] - 4s 45ms/step - loss: 1.3791 - accuracy: 0.5262
Epoch 9/30
98/98 [==============================] - 4s 45ms/step - loss: 1.2276 - accuracy: 0.5749
Epoch 10/30
98/98 [==============================] - 4s 46ms/step - loss: 1.1114 - accuracy: 0.6184
Epoch 11/30
98/98 [==============================] - 4s 46ms/step - loss: 1.0243 - accuracy: 0.6495
Epoch 12/30
98/98 [==============================] - 4s 46ms/step - loss: 0.9061 - accuracy: 0.6900
Epoch 13/30
98/98 [==============================] - 4s 46ms/step - loss: 0.8181 - accuracy: 0.7207
Epoch 14/30
98/98 [==============================] - 4s 46ms/step - loss: 0.7295 - accuracy: 0.7532
Epoch 15/30
98/98 [==============================] - 5s 47ms/step - loss: 0.6243 - accuracy: 0.7862
Epoch 16/30
98/98 [==============================] - 5s 47ms/step - loss: 0.5658 - accuracy: 0.8075
Epoch 17/30
98/98 [==============================] - 5s 46ms/step - loss: 0.4800 - accuracy: 0.8389
Epoch 18/30
98/98 [==============================] - 5s 46ms/step - loss: 0.4346 - accuracy: 0.8538
Epoch 19/30
98/98 [==============================] - 5s 46ms/step - loss: 0.3832 - accuracy: 0.8728
Epoch 20/30
98/98 [==============================] - 5s 47ms/step - loss: 0.3410 - accuracy: 0.8880
Epoch 21/30
98/98 [==============================] - 5s 47ms/step - loss: 0.3249 - accuracy: 0.8950
Epoch 22/30
98/98 [==============================] - 5s 47ms/step - loss: 0.2992 - accuracy: 0.9045
Epoch 23/30
98/98 [==============================] - 5s 47ms/step - loss: 0.2786 - accuracy: 0.9122
Epoch 24/30
98/98 [==============================] - 5s 47ms/step - loss: 0.2654 - accuracy: 0.9196
Epoch 25/30
98/98 [==============================] - 5s 47ms/step - loss: 0.2814 - accuracy: 0.9168
Epoch 26/30
98/98 [==============================] - 5s 47ms/step - loss: 0.2353 - accuracy: 0.9277
Epoch 27/30
98/98 [==============================] - 5s 47ms/step - loss: 0.2170 - accuracy: 0.9337
Epoch 28/30
98/98 [==============================] - 5s 47ms/step - loss: 0.2348 - accuracy: 0.9324
Epoch 29/30
98/98 [==============================] - 5s 47ms/step - loss: 0.2117 - accuracy: 0.9384
Epoch 30/30
98/98 [==============================] - 5s 47ms/step - loss: 0.2224 - accuracy: 0.9370

score=model.evaluate(test_data, test_label)
print(score[1])
     
313/313 [==============================] - 1s 4ms/step - loss: 3.4559 - accuracy: 0.5923
0.5922999978065491

#繪製正確性與損失歷史圖
import matplotlib.pyplot as plt
import numpy as np

def show_train_history(train_history, train):
    plt.plot(train_history.history[train])
    plt.title('Train History')
    plt.ylabel(train)
    plt.xlabel('Epoch')
    plt.legend(['train'], loc='center right')
    plt.show()

show_train_history(train_history, 'accuracy')
show_train_history(train_history, 'loss')
